### PR TITLE
[Scheduler] Support per task/actor SpreadSchedulingStrategy

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
This PR adds per task/actor SpreadSchedulingStrategy which will try to spread the tasks on a best effort basis.

## Related issue number
#18524 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_raylet.pyx [hunk 1]
```diff
@@ -1459,6 +1464,7 @@ cdef class CoreWorker:
                 f"{python_scheduling_strategy}. "
                 f"Valid values are [\"DEFAULT\""
                 f" | DefaultSchedulingStrategy"
+                f" | \"SPREAD\" | SpreadSchedulingStrategy"
```
Comment (ericl) [inline python/ray/_raylet.pyx:1467]: How about we remove the DefaultSchedulingStrategy and SpreadSchedulingStrategy Python classes? Seems they don't really serve a purpose when we have the string form, and we don't accept arbitrary subclasses of SchedulingStrategy.

File: src/ray/raylet/scheduling/cluster_resource_scheduler.cc [hunk 2]
```diff
@@ -245,7 +246,12 @@ int64_t ClusterResourceScheduler::GetBestSchedulableNode(
   // TODO (Alex): Setting require_available == force_spillback is a hack in order to
   // remain bug compatible with the legacy scheduling algorithms.
   int64_t best_node_id = scheduling_policy_->HybridPolicy(
-      resource_request, force_spillback, force_spillback,
+      resource_request,
+      scheduling_strategy.scheduling_strategy_case() ==
+              rpc::SchedulingStrategy::SchedulingStrategyCase::kSpreadSchedulingStrategy
+          ? 0.0
+          : RayConfig::instance().scheduler_spread_threshold(),
+      force_spillback, force_spillback,
```
Comment (ericl) [inline src/ray/raylet/scheduling/cluster_resource_scheduler.cc:254]: Do we do round robin always in default as well?

Comment (jjyao) [reply src/ray/raylet/scheduling/cluster_resource_scheduler.cc:254]: I limit the scope of this PR to only add “SPREAD”, no round robin fix (next PR).

**Comment by jjyao** [comment]
Ready for more reviews.

**Review by ericl — APPROVED** [review]

**Comment by jjyao** [comment]
It's ready to be merged.

File: python/ray/util/scheduling_strategies.py [hunk 3]
```diff
@@ -2,26 +2,17 @@
...
-    This disables any potential placement group capture.
-    """
-    pass
+# Spread scheduling on a best effort basis.
+SPREAD_SCHEDULING_STRATEGY = "SPREAD"
 
 
 @PublicAPI(stability="beta")
-class PlacementGroupSchedulingStrategy(SchedulingStrategy):
+class PlacementGroupSchedulingStrategy(object):
```
Comment (clarkzinzow) [inline python/ray/util/scheduling_strategies.py:15]: FYI @jjyao, you don't need to subclass `object` in Python 3 to get a new-style class, and now that Python 2 has reached EOL, it's considered best practices to not subclass `object`: `class PlacementGroupSchedulingStrategy:`

Comment (jjyao) [reply python/ray/util/scheduling_strategies.py:15]: ok, I'll change it in my next PR.

**Pull request merged**